python/python/lance/dataset.py

```diff
@@ -1516,7 +1516,12 @@ def create_index(
             kwargs["shuffle_partition_concurrency"] = shuffle_partition_concurrency
 
         self._ds.create_index(column, index_type, name, replace, kwargs)
-        return LanceDataset(self.uri, index_cache_size=index_cache_size)
+
+        storage_options = None
+        if "storage_options" in kwargs:
+            storage_options = kwargs.pop("storage_options")
+
+        return LanceDataset(self.uri, index_cache_size=index_cache_size, storage_options=storage_options)
```

Comment:
I've tested this locally and it indeed works fine with `return self`. Let's just change to that.
```suggestion
        return self
```